### PR TITLE
Feature/py function names

### PR DESCRIPTION
**Pull request by heeler**
Changed functionality to have explicit scene and tile accessors for mosaic and non-mosaic files.
Changed python functions to start with verbs.
 Updated tests. 
TODO: after aicsimageio wrapper, I will update docstrings etc. Don't have time now.

### CODE REVIEW

File: _aicspylibczi/Reader.cpp
Comment (JacksonMaxfield) [inline _aicspylibczi/Reader.cpp:392]: Remove commented out lines.

File: _aicspylibczi/Reader.h
Comment (JacksonMaxfield) [inline _aicspylibczi/Reader.h:228]: Remove commented out lines.

File: _aicspylibczi/Reader.h
Comment (JacksonMaxfield) [inline _aicspylibczi/Reader.h:275]: Remove commented out lines.

File: _aicspylibczi/Reader.h
Comment (JacksonMaxfield) [inline _aicspylibczi/Reader.h:339]: what counts as a "matching" bounding box?

File: aicspylibczi/CziFile.py
Comment (JacksonMaxfield) [inline aicspylibczi/CziFile.py:191]: Nit: this is the same as above from `get_tile_bouding_box`, may want to make a `_` function to do this operation in a single place so that the two don't get out of sync if you ever need to update them / find a bug.

File: aicspylibczi/CziFile.py
Comment (JacksonMaxfield) [inline aicspylibczi/CziFile.py:482]: I think I now see that the lines I was commenting on above are all the same python one-liner. I still stand by my statement that this "one-liner" should be moved to its own function haha.

File: aicspylibczi/tests/test_czi_file.py
Comment (JacksonMaxfield) [inline aicspylibczi/tests/test_czi_file.py:167]: Remove prints.

File: aicspylibczi/tests/test_czi_file.py
Comment (JacksonMaxfield) [inline aicspylibczi/tests/test_czi_file.py:192]: Should these be uncommented or removed?

File: aicspylibczi/tests/test_czi_file.py
Comment (JacksonMaxfield) [inline aicspylibczi/tests/test_czi_file.py:521]: Remove print

File: c_tests/test_Reader.cpp
Comment (JacksonMaxfield) [inline c_tests/test_Reader.cpp:492]: Remove commented out line

File: c_tests/test_Reader.cpp
Comment (JacksonMaxfield) [inline c_tests/test_Reader.cpp:610]: Remove commented out line

**Review by JacksonMaxfield — CHANGES_REQUESTED** [review]
This all looks good. A bunch of nitpick comments.

My major comment is about the release and aicsimageio dev strategy.
`aicspylibczi` 3.0.0 needs to be released before `aicsimageio` 4.0.0 because we need to pin it as a dependency. You won't even be able to push or open a PR to aicsimageio with the updated reader until this is merged and 3.0.0 is released. Because of this. you can start working on `aicsimageio.readers.czi_reader.CZIReader` now but before _this_ PR is merged you **need** to add docstrings to this code.

So my recommendation is to finish this PR first. If waiting on Dan or someone else sure get started on `aicsimageio.readers.czi_reader.CZIReader` but this PR and non-release of 3.0.0 here limits your work on `aicsimageio`.

Also because of the changes here, we need to write a release notes for 3.0.0 that lists all the changes just like how I did with `aicsimageio` 4.0.0 release notes. I am sure some people will be confused and a bit frustrated by the sudden change without notice.

File: _aicspylibczi/Reader.h
Comment (JacksonMaxfield) [inline _aicspylibczi/Reader.h:272]: Incomplete doc?

File: _aicspylibczi/Reader.h
Comment (JacksonMaxfield) [inline _aicspylibczi/Reader.h:280]: In complete doc?

File: aicspylibczi/CziFile.py
Comment (JacksonMaxfield) [inline aicspylibczi/CziFile.py:235]: typo on `tle` -> `tile`? or is it actually called `tle`?

File: aicspylibczi/CziFile.py
Comment (JacksonMaxfield) [inline aicspylibczi/CziFile.py:370]: No need to include this if there are no params.

File: aicspylibczi/CziFile.py
Comment (JacksonMaxfield) [inline aicspylibczi/CziFile.py:253]: No need to include.

**Review by JacksonMaxfield — APPROVED** [review]
Looks good to me. Some nit picky comments about docstrings mostly. Since I don't have the time to hit the CziReader on aicsimageio this weekend, may be best to wait for @toloudis to also review.

File: aicspylibczi/CziFile.py
Comment (heeler) [reply aicspylibczi/CziFile.py]: it's returning all the scene bounding boxes it's a really compact C++ function being exposed. Is the concern that it doesn't expose the mechanics?

File: aicspylibczi/CziFile.py
Comment (heeler) [reply aicspylibczi/CziFile.py:235]: tle is just a variable name I was worried about using the word tile and a variable named tile in the same block so I mutated it. Would it be better if it were `tile.dimension_coordinates = A dictionary of Dimensions for the tile/subblock`?

Comment (JacksonMaxfield) [reply aicspylibczi/CziFile.py:235]: Yea I think it is nitpicky of me but would make more sense.

File: aicspylibczi/CziFile.py
Comment (JacksonMaxfield) [reply aicspylibczi/CziFile.py]: Yea no worries. I read through all the docstrings / functions and I _think_ I understand it all.

**Comment by heeler** [comment]
Release notes https://github.com/AllenCellModeling/aicspylibczi/releases/edit/untagged-f007b5781ae0d70926fa

File: ROADMAP.md
Comment (toloudis) [inline ROADMAP.md:31]: instead of `the A channel` it should probably say `the A dimension`... the A channel makes it sound like alpha.

File: ROADMAP.md
Comment (toloudis) [inline ROADMAP.md:43]: provide enable?

**Comment by toloudis** [comment]
My understanding of the core processing code is not great, but I am happy that it's nice and cleanly formatted and it's very easy to read as a result.
One thing that slightly scares me stylistically is if statements with no curly brackets.  Obviously it's fine but I've seen it lead to bugs when people come in and make modifications to the code.

My only other concerns are the above items for readability in the roadmap text file.

**Review by toloudis — APPROVED** [review]

**Pull request merged**
